x-pack/plugins/osquery/server/lib/telemetry/receiver.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  Logger,
+  CoreStart,
+  ElasticsearchClient,
+  SavedObjectsClientContract,
+} from 'src/core/server';
+
+import { packSavedObjectType, savedQuerySavedObjectType } from '../../../common/types';
+import { AgentClient, AgentPolicyServiceInterface } from '../../../../fleet/server';
+import type { ESLicense, ESClusterInfo } from './types';
+import { OsqueryAppContextService } from '../osquery_app_context_services';
+
+export class TelemetryReceiver {
+  private readonly logger: Logger;
+  private agentClient?: AgentClient;
+  private agentPolicyService?: AgentPolicyServiceInterface;
+  private esClient?: ElasticsearchClient;
+  private soClient?: SavedObjectsClientContract;
+  private clusterInfo?: ESClusterInfo;
+  private readonly max_records = 10_000;
```

Comment:
Thank you @afharo, I've reduced this number to 100 and removed batching logic, so we will be sending 2 requests per day per cluster (at maximum, because if there are no packs/saved queries no data will be sent)